### PR TITLE
Add first flare e2e tests

### PR DESCRIPTION
**Pull request by Kaderinho**
### What does this PR do?

This PR adds tests for the flare subcommand

### Additional Notes

It should be merged after https://github.com/DataDog/datadog-agent/pull/18149

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: test/new-e2e/agent-subcommands/flare/flare_helpers.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+// isDir returns true if a `filename` is a directory
+func isDir(flare flare.Flare, filename string) bool {
+	fileInfo, err := flare.GetFileInfo(filename)
+	if err != nil {
+		return false
+	}
+	return fileInfo.IsDir()
+}
+
+func assertProcessCheckShouldBeEnabled(t *testing.T, flare flare.Flare, checkName string, setting string, shouldBeEnabled bool) {
```
Comment (Kaderinho) [inline test/new-e2e/agent-subcommands/flare/flare_helpers.go:100]: Add comment for functions below

File: test/new-e2e/agent-subcommands/flare/flare_files.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	"runtime_config_dump.yaml",
+	"secrets.log",
+	"status.log",
+	"system_probe_runtime_config_dump.yaml",
+	"tagger-list.json",
+	"version-history.json",
+	"workload-list.log",
+}
+
+var defaultLogFiles = []string{
```
Comment (Kaderinho) [inline test/new-e2e/agent-subcommands/flare/flare_files.go:32]: Add comments for all files list

File: test/new-e2e/agent-subcommands/flare/flare_test.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	v.Env().VM.Execute(fmt.Sprintf(`echo "%s" | sudo tee /etc/datadog-agent/security-agent.yaml`, securityAgentConfiguration))
+	v.UpdateEnv(e2e.AgentStackDef(nil, agentparams.WithAgentConfig(string(agentConfiguration))))
+
+	err := v.Env().Agent.WaitForReady()
+	assert.NoError(v.T(), err)
+
+	_ = v.Env().Agent.Flare(client.WithArgs("--email e2e@test.com --send"))
+
+	flare, err := v.Env().Fakeintake.Client.GetLatestFlare()
+	assert.NoError(v.T(), err)
```
Comment (Kaderinho) [inline test/new-e2e/agent-subcommands/flare/flare_test.go:85]: Use `waitForAgentAndGetFlare` instead

File: test/new-e2e/agent-subcommands/flare/flare_test.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+type commandFlareSuite struct {
+	e2e.Suite[e2e.AgentEnv]
+}
+
+func TestFlareSuite(t *testing.T) {
+	e2e.Run(t, &commandFlareSuite{}, e2e.AgentStackDef(nil), params.WithDevMode())
+}
+
+func waitForAgentAndGetFlare(v *commandFlareSuite, flareArgs ...client.AgentArgsOption) flare.Flare {
+	err := v.Env().Agent.WaitForReady()
```
Comment (Kaderinho) [inline test/new-e2e/agent-subcommands/flare/flare_test.go:30]: Should not be necessary anymore

**Review by pgimalac — APPROVED** [review]

File: test/new-e2e/agent-subcommands/flare/flare_helpers.go [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+import (
+	"strings"
+	"testing"
+
+	"github.com/DataDog/datadog-agent/test/fakeintake/client/flare"
+	"github.com/stretchr/testify/assert"
+)
+
+// assertFilesExist verifies that all files in filenames exist in the flare archive
+func assertFilesExist(t *testing.T, flare flare.Flare, filenames []string) {
```
[the remark this answers is not in the document]
Comment (Kaderinho) [reply test/new-e2e/agent-subcommands/flare/flare_helpers.go:17]: TIL, thanks ! Done here 748e89a34cd6a76d2a9db7bc6ead3a15a8634a7e

File: test/new-e2e/agent-subcommands/flare/flare_test.go [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+		"telemetry.log",       // if telemetry.enabled
+		"registry.json",       // if Logs Agent is running
+		"expvar/system-probe", // if system probe is enabled
+	}
+
+	// XXX: use FileManager here instead
+	// XXX: this test is expected to fail because 'etc/security-agent.yaml' is not found. See #18463
+	// TODO: use dedicated functions when https://github.com/DataDog/test-infra-definitions/pull/309 is merged
+	v.Env().VM.Execute(fmt.Sprintf(`echo "%s" | sudo tee /etc/datadog-agent/system-probe.yaml`, systemProbeConfiguration))
+	v.Env().VM.Execute(fmt.Sprintf(`echo "%s" | sudo tee /etc/datadog-agent/security-agent.yaml`, securityAgentConfiguration))
```
[the remark this answers is not in the document]
Comment (Kaderinho) [reply test/new-e2e/agent-subcommands/flare/flare_test.go]: Yup, you're right 77fb23c5ae8e62402d11acb609f6843d9d92167a

File: test/new-e2e/agent-subcommands/flare/flare_test.go [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+	"github.com/stretchr/testify/assert"
+)
+
+type commandFlareSuite struct {
+	e2e.Suite[e2e.FakeIntakeEnv]
+}
+
+func TestFlareSuite(t *testing.T) {
+	e2e.Run(t, &commandFlareSuite{}, e2e.FakeIntakeStackDef(nil), params.WithDevMode())
```
Comment (pducolin) [inline test/new-e2e/agent-subcommands/flare/flare_test.go:27]: **🔨 warning**
You should pass `params.WithDevMode` in local runs. You could have a slice of options and append `WithDevMode` only when `CI` env var is false

File: test/new-e2e/agent-subcommands/flare/flare_test.go [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+
+	flare, err := v.Env().Fakeintake.Client.GetLatestFlare()
+	assert.NoError(v.T(), err)
+
+	return flare
+}
+
+func (v *commandFlareSuite) TestFlareDefaultFiles() {
+	v.UpdateEnv(e2e.FakeIntakeStackDef(nil))
+	flare := waitForAgentAndGetFlare(v, client.WithArgs([]string{"--email", "e2e@test.com", "--send"}))
```
Comment (pducolin) [inline test/new-e2e/agent-subcommands/flare/flare_test.go:41]: **💭 thought**
For readability, I would rather act and then collect data, in other words I would call

```golang

// act
v.Env().Agent.Flare(..)

// assert
flare, err := v.Env().Fakeintake.Client.GetLatestFlareEventually()
...
```

File: test/new-e2e/agent-subcommands/flare/flare_test.go (hunk 7 above)
Comment (Kaderinho) [reply test/new-e2e/agent-subcommands/flare/flare_test.go:27]: I plan to remove it when I got all needed approvals

File: test/new-e2e/agent-subcommands/flare/flare_test.go (hunk 8 above)
Comment (Kaderinho) [reply test/new-e2e/agent-subcommands/flare/flare_test.go:41]: I don't get your point sorry, do you want me to not call a function and directly call `Flare()` and `GetLatestFlare()` in each test scenario ?

File: test/new-e2e/agent-subcommands/flare/flare_test.go (hunk 7 above)
Comment (Kaderinho) [reply test/new-e2e/agent-subcommands/flare/flare_test.go:27]: > You could have a slice of options and append WithDevMode only when CI env var is false

Could it be something added directly into the framework ?

**Review by pducolin — APPROVED** [review]

File: test/new-e2e/agent-subcommands/flare/flare_test.go (hunk 7 above)
Comment (pducolin) [reply test/new-e2e/agent-subcommands/flare/flare_test.go:27]: Yes, absolutely !

**Pull request merged**